RELEASES.md

```diff
@@ -1,5 +1,14 @@
 # Releases
 
+## 18.3.6 (March 12th, 2025)
+
+Version 17.x is now being maintained / released from the [release17.x branch](https://github.com/microsoft/applicationinsights-react-js/tree/release17.3.x) as
```

Comment:
Not quite, I listed the correct branches in the main readme.

Lets just say that this release is targeting React 18 and is using Application Insights 3.3.6 and that we are bumping the major version number to match the framework, basically following the same numbering as outlined here https://github.com/microsoft/applicationinsights-angularplugin-js/issues/57

Including the branch names here is going to become outdated quickly, maybe just link to the compatibility matrix instead.